### PR TITLE
[AutoDiff] TF-189: Fix pullback type calculation.

### PR DESCRIPTION
**Pull request by rxwei**
The cotangent type may not be of the same triviality as the corresponding original type. Fix that.

Resolves [TF-189](https://bugs.swift.org/browse/TF-189).

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

File: lib/SIL/SILFunctionType.cpp [hunk 1]
```diff
@@ -329,7 +336,7 @@ CanSILFunctionType SILFunctionType::getAutoDiffAssociatedFunctionType(
                 AutoDiffAssociatedVectorSpaceKind::Cotangent, lookupConformance)
             ->getCanonicalType();
     cotangentParams.push_back(
-        getParameterInfoForOriginalResult(cotangentAssocTy,
+        getCotangentParameterInfoForOriginalResult(cotangentAssocTy,
                                           origRes.getConvention()));
```
Comment (rxwei) [inline lib/SIL/SILFunctionType.cpp]: Noted indentation mismatch. Will fix next time.

**Review by marcrasi — APPROVED** [review]

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Review by dan-zheng — APPROVED** [review]

**Pull request merged**